libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -35,32 +36,89 @@ pub struct Error {
     pub msg: String,
 }
 
-/// DocPackage represents the documentation for a package and its sub packages
+/// Doc is an enum that can take the form of any form of flux documentation through polymorphism.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DocPackage {
-    /// the path to the documentation package
-    pub path: String,
-    /// the name of the comments package?
+pub enum Doc {
+    /// Package represents documentation for an entire Flux package.
+    Package(Box<PackageDoc>),
+    /// Value represents documentation for a value exposed from a package.
+    Value(Box<ValueDoc>),
+    /// Builtin represents documentation for a builtin value exposed from a package.
+    Builtin(Box<ValueDoc>),
+    /// Option represents documentation for a option value exposed from a package.
+    Option(Box<ValueDoc>),
+    /// Function represents documentation for a function value exposed from a package.
+    Function(Box<FunctionDoc>),
+}
+
+/// PackageDoc represents the documentation for a package and its sub packages
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PackageDoc {
+    /// the name of the comments package
     pub name: String,
-    /// the doc
-    pub doc: String,
-    /// the values of the comments
-    pub values: Vec<DocValue>,
+    /// the headline of the package
+    pub headline: String,
+    /// the description of the package
+    pub description: Option<String>,
+    /// the members of the package which are values and functions with their parameters
```

Comment:
You could say something like "The values and functions of a package" here, and that would tell me all I need.